src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheck.java

```diff
@@ -0,0 +1,101 @@
+package org.openstreetmap.atlas.checks.validation.tag;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+import org.openstreetmap.atlas.utilities.scalars.Distance;
+
+/**
+ * Flags road segments that have no {@link org.openstreetmap.atlas.tags.names.NameTag} or a NameTag
+ * with a different spelling from other segments of the same road. This check is primarily meant to
+ * catch small errors in spelling, such as a missing letter or letter accent mixups.
+ *
+ * @author seancoulter
+ */
+public class RoadNameSpellingConsistencyCheck extends BaseCheck<Long>
+{
+
+    private static final long serialVersionUID = 1L;
+
+    private static final HighwayTag MINIMUM_NAME_PRIORITY_DEFAULT = HighwayTag.SERVICE;
+    private static final double MAXIMUM_SEARCH_DISTANCE_DEFAULT = 300.0;
+    private static final double MAXIMUM_CHARACTER_DIFFERENCES_DEFAULT = 1;
+    private static final String FALLBACK_INSTRUCTIONS = "These road segments have spelling inconsistencies. One spelling is: %s. Examine all flagged road segments to determine the best spelling, and apply this spelling to all of those segments.";
+
+    private final Distance maximumSearchDistance;
+    private final double inconsistentCharacterCountThreshold;
+
+    public RoadNameSpellingConsistencyCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.maximumSearchDistance = this.configurationValue(configuration,
+                "distance.search.maximum", MAXIMUM_SEARCH_DISTANCE_DEFAULT, Distance::meters);
+        this.inconsistentCharacterCountThreshold = this.configurationValue(configuration,
+                "characterCount.inconsistencies", MAXIMUM_CHARACTER_DIFFERENCES_DEFAULT);
+    }
+
+    /**
+     * This function will validate if the supplied atlas object is valid for the check.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return object instanceof Edge
+                && ((Edge) object).highwayTag().isMoreImportantThanOrEqualTo(
+                        MINIMUM_NAME_PRIORITY_DEFAULT)
+                && ((Edge) object).isMasterEdge() && !this.isFlagged(object.getIdentifier())
+                && ((Edge) object).getName().isPresent();
```

Comment:
The requirements are self-imposed so far, so we could definitely look at the other tags.I figured we'd only want to flag the most "visible" (e.g. can see in the editor) names, but I'm open to your idea. Maybe it could be a configurable list of name related tags to flag

Reply:
I'm a big fan of potentially looking at other names!

One possible concern with this, though, is that edges can have multiple names, which might differ by one character in valid ways. For each rule I can come up to handle this, I can come up with a counterexample where that rule doesn't make sense. Maybe there's something I haven't thought of, though :)